### PR TITLE
Shuffle on K8s

### PR DESCRIPTION
**Pull request by dhaval055**

### CODE REVIEW

**Comment by frikky** [comment]
👀

File: functions/kubernetes/kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Step 5: Create storage-class for shuffle by running ```kubectl apply –f shuffle-sc.yaml –n shuffle```.  
+
+Step 6: Apply the shuffle-opensearch.yaml configuration to create a Persistent Volume Claim (PVC) for storage, a Deployment to manage the OpenSearch pod, and a Service to expose OpenSearch within the cluster by using  ```kubectl apply –f shuffle-opensearch.yaml  –n shuffle```.
+
+Step 7: Now deploy the backend using ```kubectl apply –f shuffle-backend.yaml  –n shuffle```.
+
+Step 8: Ensure that the Shuffle Backend is successfully deployed and running before proceeding with the frontend deployment, as the frontend may depend on services provided by the backend. ```kubectl apply –f shuffle-frontend.yaml –n shuffle```.
+
+Step 9: Run ```kubectl apply -f shuffle-orborus.yaml```. This final step ensures that Shuffle Orborus is deployed, enabling the workflow executions in your Kubernetes-based Shuffle setup.
```
Comment (0x0elliot) [inline functions/kubernetes/kubernetes.md:24]: Is there a reason shuffle-orborus is outside the shuffle namespace?

**Review by 0x0elliot — COMMENTED** [review]
Tried following this guide. I am not that familiar with k8s, so that was my perspective.

Screenshot 2023-10-16 at 2 41 53 PM

File: functions/kubernetes/kubernetes.md (hunk 1 above)
Comment (dhaval055) [reply functions/kubernetes/kubernetes.md:24]: > Is there a reason shuffle-orborus is outside the shuffle namespace?

No, it should be inside the shuffle namespace. error in the docs. it should be 'kubectl apply -f shuffle-orborus.yaml -n shuffle'

Comment (0x0elliot) [reply functions/kubernetes/kubernetes.md:24]: gotcha.

**Comment by 0x0elliot** [comment]
i will test this further by EOD and give more inputs.

File: functions/kubernetes/kubernetes.md [hunk 2]
```diff
@@ -3,26 +3,42 @@
 ### Prerequisites:
 -	Clone the https://github.com/dhaval055/shuffle repository using Git and change the branch to shuffle-k8s then navigate to the functions/kubernetes directory, which contains all the necessary Kubernetes configuration files for deployment.
 -	Running Kubernetes cluster.
-- Ensure you have a local Docker registry set up to store and manage Docker images for applications built with Shuffle. While the registry is crucial for handling custom-built apps, you’ll still be able to run workflows without it. 
-
+- Ensure you have a local Docker registry set up to store and manage Docker images for applications built with Shuffle. While the registry is crucial for handling custom-built apps, you’ll still be able to run workflows without it. To setup a docker registry, if you have docker installed on one of your node run following commands.
```
Comment (0x0elliot) [inline functions/kubernetes/kubernetes.md:6]: leaving here for now, assumption is: the customer gets a guide to copy paste commands from. i know it's annoying but add in: `mkdir certs` as well.

if you don't, i will do it later so don't worry.

File: functions/kubernetes/kubernetes.md [hunk 3]
```diff
@@ -3,26 +3,42 @@
...
+  ```
+  openssl req -newkey rsa:4096 -nodes -sha256 -keyout certs/reg.key -x509 -days 365 -out certs/reg.crt -subj "/CN=<YOUR_NODE_LOCAL_IP>"
```
Comment (0x0elliot) [inline functions/kubernetes/kubernetes.md:8]: ideally, we should find a way to automate this. 

do you think we can do that?

**Comment by 0x0elliot** [comment]
got everything running instead of `opensearch-6488f65c45-hwnld`. t's stuck on "pending".

Screenshot 2023-10-16 at 11 59 24 PM
Screenshot 2023-10-17 at 12 00 40 AM

**Comment by 0x0elliot** [comment]
probably something to do with: https://stackoverflow.com/questions/67817865/default-scheduler-0-1-nodes-are-available-1-nodes-didnt-find-available-persi ?

**Comment by dhaval055** [comment]
> probably something to do with: https://stackoverflow.com/questions/67817865/default-scheduler-0-1-nodes-are-available-1-nodes-didnt-find-available-persi ?

Please run 'kubectl describe pvc opensearch-claim0' and then 'kubectl describe pv shuffle-os-pv' to see whats wrong.

**Pull request merged**